Mathlib/NumberTheory/LSeries/PrimesInAP.lean

```diff
@@ -15,12 +15,52 @@ The goal of this file is to prove **Dirichlet's Theorem**: If `q` is a positive
 and `a : ZMod q` is invertible, then there are infinitely many prime numbers `p` such that
 `(p : ZMod q) = a`.
 
-This will be done in the following steps:
-
-1. Some auxiliary lemmas on infinite products and sums
-2. Results on the von Mangoldt function restricted to a residue class
-3. (TODO) Results on its L-series and an auxiliary function related to it
-4. (TODO) Derivation of Dirichlet's Theorem
+The main steps of the proof are as follows.
+1. Define `ArithmeticFunction.vonMangoldt.residueClass a` for `a : ZMod q`, which is
+   a function `ℕ → ℝ` taking the value zero when `(n : ℤMod q) ≠ a` and `Λ n` else
+   (where `Λ` is the von Mangoldt function `ArithmeticFunction.vonMangoldt`; we have
+   `Λ (p^k) = log p` for prime powers and `Λ n = 0` otherwise.)
+2. Show that this function can be written as a linear combination of functions
+   of the form `χ * Λ` (pointwise product) with Dirichlet characters `χ` mod `q`.
+   See `ArithmeticFunction.vonMangoldt.residueClass_eq`.
+3. This implies that the L-series of `ArithmeticFunction.vonMangoldt.residueClass a`
+   agrees (on `re s > 1`) with the corresponding linear combination of negative logarithmic
+   derivatives of Dirichlet L-functions.
+   See `ArithmeticFunction.vonMangoldt.LSeries_residueClass_eq`.
+4. Define an auxiliary function `ArithmeticFunction.vonMangoldt.auxFun a` that is
```

Comment:
It seems that `auxFun` plays quite an important role. I think it would be good if we can give it a more descriptive name. But I recognize that this is maybe very hard.

Do you have a suggestion? Or shall we ask on zulip for suggestions?

Reply:
`Lfunction_residueClass_sub_inv_totient_div`? :smile: (It's a bit long...)

Reply:
Changed to `LfunctionResidueClassAux` as per @adomani's suggestion on  Zulip.